components/bounty/page-actions/view.tsx

```diff
@@ -0,0 +1,289 @@
+          </ContractButton>
+        </ReadOnlyButtonWrapper>
+      );
+  }
+
+  function renderCreateProposalButton() {
+    if (isCreateProposal)
+      return (
+        <ReadOnlyButtonWrapper>
+          <ContractButton
+            className="read-only-button bounty-outline-button"
+            onClick={() => handleShowPRModal(true)}
+            disabled={!currentUser?.login || !isWalletConnected}
+          >
+            <Translation ns="proposal" label="actions.create.title"/>
+          </ContractButton>
+        </ReadOnlyButtonWrapper>
+      );
+  }
+
+  function renderEditButton() {
+    if (isWalletConnected && isBountyInDraft && isBountyOwner)
+      return (
+        <ReadOnlyButtonWrapper>
+          <ContractButton
+            className="read-only-button bounty-outline-button me-1"
+            onClick={handleEditIssue}
+          >
+            <EditIcon className="me-1"/>
+            <Translation ns="bounty" label="actions.edit-bounty" />
+          </ContractButton>
+        </ReadOnlyButtonWrapper>
+      );
+  }
+
+  function renderTableAndMobileButton() {
+    const actions: Action[] = []
+
+    if (isCreatePr)
+      actions.push({
+        label: "Pull Request",
+        onClick: () => handleShowPRModal(true),
+      });
+      
+    if (isCreateProposal)
+      actions.push({
+        label: "Proposal",
+        onClick: () => setShowPRProposal(true),
+      });
+
+    if (!isGithubConnected && isWalletConnected)
+      return <ConnectGithub size="lg" />;
+
+    if (isCreatePr || isCreateProposal)
+      return (
+          <MultiActionButton
+            label="Create"
+            className="col-12"
+            actions={actions}
+          />
+      );
+
+    return renderStartWorkingButton();
+  }
+
+  return (
+    <div className="container mt-4">
+      <div className="row justify-content-center">
+        <div className="col-md-12">
+          {(!isGithubConnected && isWalletConnected) && 
+          <ContextualSpan context="info" className="mb-2" isAlert>
+            {t("actions.connect-github-to-work")}
+          </ContextualSpan>}
+
+          <div className="d-flex align-items-center justify-content-between mb-4">
+            <h4 className="h4 d-flex align-items-center d-none d-lg-block">
+              {t("misc.details")}
+            </h4>
+
+            <div className="d-flex flex-row align-items-center gap-20 d-none d-lg-block">
+              {renderForkRepositoryLink()}
+
+              {renderStartWorkingButton()}
+
+              {renderCreatePullRequestButton()}
+
+              {renderUpdateAmountButton()}
+
+              {renderCreateProposalButton()}
+
+              {renderEditButton()}
+
+              {!isGithubConnected && isWalletConnected && <ConnectGithub size="sm"/>}
+
+            </div>
+            <div className="col-12 d-lg-none"> 
+              {renderTableAndMobileButton()}
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <BountyEffectsProvider>
```

Comment:
we shouldn't need this effect-provider because we should be handed down the data from the parent-controller and thus we should just re-render the page when needed [when needed = when we changed the bounty, etc, etc]